iroh-net/src/endpoint.rs

```diff
@@ -22,6 +22,78 @@ use crate::{
 /// How long we wait at most for some endpoints to be discovered.
 const ENDPOINT_WAIT: Duration = Duration::from_secs(5);
 
+/// Builder for [MagicEndpoint]
+#[derive(Debug, Default)]
+pub struct MagicEndpointBuilder {
+    keypair: Option<Keypair>,
+    derp_map: Option<DerpMap>,
+    alpn_protocols: Vec<Vec<u8>>,
+    transport_config: Option<quinn::TransportConfig>,
+    keylog: bool,
+}
+
+impl MagicEndpointBuilder {
+    /// Set a keypair to authenticate with other peers.
+    ///
+    /// This keypair's public key will be the [PeerId] of this endpoint.
+    ///
+    /// If not set, a new keypair will be generated.
+    pub fn keypair(self, keypair: Keypair) -> Self {
+        Self {
+            keypair: Some(keypair),
+            ..self
```

Comment:
interesting way to write this, any reason to use this over?

```rust
self.keypair.replace(keypair);
self
```

Reply:
no reason, changed it